src/lib.rs

```diff
@@ -105,11 +127,141 @@ impl Fairing for RocketSentry {
                 if config.sentry_dsn.is_empty() {
                     info!("Sentry disabled.");
                 } else {
-                    self.init(&config.sentry_dsn);
+                    let traces_sample_rate = config.sentry_transaction_sample_rate.unwrap_or(0f32);
+                    self.init(&config.sentry_dsn, traces_sample_rate);
                 }
             }
             Err(err) => error!("Sentry not configured: {}", err),
         }
         Ok(rocket)
     }
+
+    async fn on_request(&self, request: &mut Request<'_>, _: &mut Data<'_>) {
+        let name = request_to_transaction_name(request);
+        let build_transaction = move || Self::start_transaction(&name);
+        let request_transaction = local_cache_once!(request, build_transaction);
+        request.local_cache(request_transaction);
+    }
+
+    async fn on_response<'r>(&self, request: &'r Request<'_>, response: &mut Response<'r>) {
+        // We take the transaction set in the on_request callback
+        let request_transaction = local_cache_once!(request, Self::invalid_transaction);
+        let ongoing_transaction: &Transaction = request.local_cache(request_transaction);
+        ongoing_transaction.set_status(map_status(response.status()));
+        set_transaction_request(ongoing_transaction, request);
+        ongoing_transaction.clone().finish();
+    }
+}
+
+fn set_transaction_request(transaction: &Transaction, request: &Request) {
+    transaction.set_request(protocol::Request {
+        url: None,
+        method: Some(String::from(request.method().as_str())),
+        data: None,
+        query_string: request_to_query_string(request),
+        cookies: None,
+        headers: Default::default(),
+        env: Default::default(),
+    });
+}
+
+fn request_to_transaction_name(request: &Request) -> String {
+    let method = request.method();
+    let path = request.uri().path();
+    format!("{method} {path}")
+}
+
+fn request_to_query_string(request: &Request) -> Option<String> {
+    let query_string = request.uri().query()?.as_str().to_string();
+    Some(query_string)
```

Comment:
Seems cleaner to elide the temporary

```suggestion
    Some(request.uri().query()?.to_string())
```